monitoring/uss_qualifier/scd/simulator/utils.py

```diff
@@ -0,0 +1,104 @@
+
+class OutputSubDirectories(ImplicitDict):
+    ''' A class to hold information about output directories that will be generated when writing the files to disk. '''
+    injection_payloads: Path 
+    scd_rules: Path
+
+class InjectionStatus(str, enum.Enum):
+    ''' A enum to hold results of flight processing as defined by the SCD test API '''
+    Planned = 'Planned'
+    Rejected = 'Rejected'
+    ConflictWithFlight = 'ConflictWithFlight'
+    Failed = 'Failed'
+
+## Definitions around operational intent data that needs to be submitted to the test injection interface
+
+class GeometryGenerationRule(ImplicitDict):
+    """ A class to hold configuration for generation of geometries """
+    intersect_space:bool = 0 # This flag specifies if the genrated geometry should interect with the first one (used only if )
+    
+class VolumeGenerationRule(ImplicitDict):
+    """ A class to hold configuration rules for flight volume generation """
+    intersect_altitude: bool = 0 # The simulator has the ability to generate multiple volumes, this is a flag to ensure that a intersection in altitude must be ensured
+    intersect_time: bool = 0 # The simulator has the ability to generate multiple volumes, this is a flag to ensure that a intersection of flight paths in time is ensured 
+    is_control: bool = 0 # When multiple volumes are generated the first one is called control against which the intersection rules are applied. 
+    expected_result: InjectionStatus.ConflictWithFlight # If the flag for intersection is set then the 
+    
+class GeneratedGeometry(ImplicitDict):
+    ''' An object to hold generated flight path, is_control is a nomenclature used to see if the generated path is the first one (against which the geometry and vole generation rules are applied) '''
+    geometry: Union[LineString, Polygon]
+    is_control: bool
+    geometry_generation_rule: GeometryGenerationRule
+  
+class GeneratedFlightPlan(ImplicitDict):
+    ''' A class used internally in the simulator with altiude and time details and GeoJSON as a flight plan the GeoJSON flight plan is converted in to a Volume 4D object and written to disk subsequently '''
+    flight_plan: Feature 
+    time_start: StringBasedDateTime
+    time_end: StringBasedDateTime
+    altitude_lower: Altitude
+    altitude_upper: Altitude
+    
+class OperationalIntentInjection(ImplicitDict):
+    ''' A class to hold data for operational intent data that will be submitted to the SCD testing interface.  For more information see : https://github.com/interuss/dss/blob/master/interfaces/automated-testing/scd/scd.yaml#L286'''
+    state: Literal[OperationalIntentState.Accepted,OperationalIntentState.Activated,OperationalIntentState.Nonconforming, OperationalIntentState.Contingent] 
+    priorty: int = 0
+    volumes: List[Volume4D]
+    off_nominal_volumes: List[Volume4D]= []
+
+
+## Definitions around flight authorization data that needs to be submitted to the test injection interface
```

Comment:
As commented above, let's put these data types in monitorlib/scd_automated_testing, like monitorlib/rid_automated_testing

Reply:
OK I will move these ones to the newly created file as well.